scripts/results_generation/results_gmcc_metric.py

```diff
@@ -0,0 +1,254 @@
+
+    :param traj_1: first trajectory to compare
+    :param traj_2: second trajectory to compare
+    :return: if first trajectory is longer than second one
+    """
+    return np.shape(traj_1)[0] > np.shape(traj_2)[0]
+
+
+def get_motion_onset_idx(trajectory: np.ndarray, tol: float) -> int:
+    """
+    Get index at which motion onset occurs
+
+    :param trajectory: data
+    :param tol: tolerance to assess motion onset
+    :return: index at which motion starts
+    """
+    nb_samples = np.shape(trajectory)[0]
+    dims = np.shape(trajectory)[1]
+    list_idx = []
+    # Loop on each dimension
+    for i in range(dims):
+        j = 0
+        while (abs(trajectory[j, i] - trajectory[j+1, i]) < tol):
+            j = j+1
+            if j == nb_samples - 1:
+                break
+        list_idx.append(j)
+    # Take lowest index as some dimensions might not be affected by motion
+    return min(list_idx)
+
+
+def pad_end(trajectory: np.ndarray, final_len: int) -> np.ndarray:
+    """
+    Pad end of trajectory
+
+    :param trajectory: data
+    :param final_len: final trajectory length
+    :return: padded trajectory
+    """
+    nb_samples = final_len - np.shape(trajectory)[0]
+    nb_dims = np.shape(trajectory)[1]
+    extension = np.full_like(trajectory, trajectory[-1, :], shape=(nb_samples, nb_dims))
+    return np.row_stack([trajectory, extension])
+
+
+def comparison_setup(regression: np.ndarray, reproduction: np.ndarray) -> list[
+        np.ndarray]:
+    """
+    Manipulate regression and reproduction trajectories to bring them at same length
+    1. Trim initial static portion of both trajectories, not moving data
+    2. Check which trajectory is longer
+    3. Pad shorter trajectory to same length
+
+    :param regression: regression trajectory
+    :param reproduction: reproduction trajectory
+    :return: list with processed trajectories ready to be compared
+    """
+    idx_regre = get_motion_onset_idx(regression, tol=10**(-1))
+    idx_repro = get_motion_onset_idx(reproduction, tol=10**(-1))
+    trimmed_regression = regression[idx_regre:, :]
+    trimmed_reproduction = reproduction[idx_repro:, :]
+    if first_is_longer(trimmed_regression, trimmed_reproduction):
+        final_len = np.shape(trimmed_regression)[0]
+        trimmed_reproduction = pad_end(trimmed_reproduction, final_len)
+    else:
+        final_len = np.shape(trimmed_reproduction)[0]
+        trimmed_regression = pad_end(trimmed_regression, final_len)
+    return [trimmed_regression, trimmed_reproduction]
+
+
+def get_gmcc_score(regression_path: str, reproduction_path: str,
+                   data_type: str) -> float:
+    """_summary_
+
+    :param regression_path: regression data path
+    :param reproduction_path: reproduction data path
+    :param data_type: tcp or joint data
+    :return: gmcc score
+    """
+    file_extension = os.path.splitext(regression_path)[1]
+    if file_extension == ".json":
```

Comment:
shouldn't that be always false?

`regression_path` is a npy and `reproduction_path` is a json?